apiserver/common/storagecommon/storage.go

```diff
@@ -241,7 +241,7 @@ func volumeAttachmentDevicePath(
 		// Prefer the volume attachment's information over what is
 		// in the published block device information.
 		var deviceLinks []string
-		if volumeAttachmentInfo.DeviceLink != "" {
+		if volumeAttachmentInfo.DeviceLink != "" && len(blockDevice.DeviceLinks) > 0 {
```

Comment:
So, this is only using volumeAttachmentInfo.DeviceLink if the block device has device links?

Reply:
Ah, from reading below it sounds like blockDevice comes from the udevadm output, so empty device links is used to detect the weird case where the output udevadm output is nearly empty, thus forcing the matching to fall back to name, is that right? Probably worth a comment if so.